lib/grape/attack/counter.rb

```diff
@@ -10,17 +10,24 @@ def initialize(request, adapter)
       end
 
       def value
-        @value ||= begin
-          adapter.get(key).to_i
-        rescue ::Grape::Attack::StoreError
-          1
+        val, exp = fetch
+
+        if exp > Time.now.to_i
+          val
+        else
+          store(0, ttl_in_seconds.seconds.from_now.to_i)
```

Comment:
I'm afraid this statement violates single responsibility principle. The `value` method is supposed to get some value whereas here depending on some condition, we either get or store some value.